vulture_os/toolkit/api_parser/waf_cloud_protector/waf_cloud_protector.py

```diff
@@ -0,0 +1,322 @@
+
+        # Needed to prevent error : _csv.Error: field larger than field limit (131072)
+        csv.field_size_limit(sys_maxsize)   
+        r = csv.reader(stream_line, delimiter=",", doublequote=True, lineterminator="\n", quoting=csv.QUOTE_ALL)
+        # Loop other the fields of the only line
+        try:
+            for cpt, value in enumerate(list(r)[0]):
+                field = mapping[cpt]
+                if field in self.TO_SPLIT_FIELDS:
+                    result[field] = {}
+                    for under_field in value.split(' and '):
+                        # Change the format of the field to be able to use it in the json 
+                        # From -> "module_name == 'xxx' and event.desc == 'xxx ' and tokens['http_xxx'] == 'xxx'"
+                        # To -> "module_name": "xxx", "event.desc": "xxx", "tokens.http_xxx": "xxx"
+                        under_field_values = [under_field_value.replace("['", ".").replace("']", "").replace("'", "") for under_field_value in  under_field.split(' == ')]
+                        if len(under_field_values) > 1:
+                            # If the field is a key value pair, we add the value to the key
+                            # It's for : "event.desc": "xxx", "tokens.http_xxx": "xxx"
+                            if '.' in under_field_values[0]:
+                                double_under_field_values = under_field_values[0].split('.')
+                                if  double_under_field_values[0] not in result[field]:
+                                    result[field][double_under_field_values[0]] = {}
+                                result[field][double_under_field_values[0]][double_under_field_values[1]] = under_field_values[1]
+                            # It's for : "module_name": "xxx"
+                            else:
+                                result[field][under_field_values[0]] = under_field_values[1]
+                        # If the field is not a key value pair, we just add the value to the key                           
+                        else:
+                            result[field][under_field_values[0]] = under_field_values[0]
+                else :
+                    result[field] = value
+            return json_dumps(result)
+        except Exception as e:
+            msg = f"Failed to parse line: {r}"
+            logger.error(f"[{__parser__}]:parse_line: {msg}", extra={'frontend': str(self.frontend)})
+            raise WAFCloudProtectorAPIError(msg)
+
+    def execute(self):
+
+        for server in self.waf_cloud_protector_servers.split(','):
+
+            # Init the timestamp for the server if not exist
+            if server not in self.frontend.waf_cloud_protector_timestamps.get('alert'):                
+                self.frontend.waf_cloud_protector_timestamps['alert'][server] = ""
+                self.frontend.waf_cloud_protector_timestamps['traffic'][server] = ""
+                self.frontend.save()
```

Comment:
Please remove this block, this can create conflicting types between string "" and datetime object